tests/by-util/test_logname.rs

```diff
@@ -29,3 +29,19 @@ fn test_normal() {
     result.success();
     assert!(!result.stdout_str().trim().is_empty());
 }
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_no_login_name() {
+    use uutests::util::TestScenario;
+    use uutests::util_name;
+    let ts = TestScenario::new(util_name!());
+
+    ts.cmd("unshare")
```

Comment:
why unshare here ?

Reply:
I've tried making getlogin return null by  
1. setting stdin to /dev/null
2. using `terminal_sim_stdio` with  `stdin` set to `false`
3. using `setsid logname`

On my machine, none of these caused it to return null. Running it with `unshare -U logname` did produce the expected error.

Do you know a more reliable way to trigger this behavior without using unshare?